apps/arweave/src/ar_pool.erl

```diff
@@ -0,0 +1,764 @@
+					{spora_2_6, MiningAddress}, not_set}) of
+		error ->
+			?LOG_ERROR([{event, pool_failed_to_validate_proof_of_access}]),
+			#partial_solution_response{ status = <<"rejected_bad_poa">> };
+		false ->
+			#partial_solution_response{ status = <<"rejected_bad_poa">> };
+		{true, ChunkID} when H1 == SolutionH ->
+			PoACache = {{BlockStart1, RecallByte1, TXRoot1, BlockSize1,
+					{spora_2_6, MiningAddress}}, ChunkID},
+			PoA2Cache = undefined,
+			process_partial_solution_difficulty(Solution, Ref, PoACache, PoA2Cache);
+		{true, ChunkID} ->
+			ComputedRecallByte2 = RecallRange2Start + Nonce * ?DATA_CHUNK_SIZE,
+			{BlockStart2, BlockEnd2, TXRoot2} = ar_block_index:get_block_bounds(
+					ComputedRecallByte2),
+			BlockSize2 = BlockEnd2 - BlockStart2,
+			case RecallByte2 == ComputedRecallByte2 andalso
+					ar_poa:validate({BlockStart2, RecallByte2, TXRoot2, BlockSize2,
+									PoA2, {spora_2_6, MiningAddress}, not_set}) of
+				error ->
+					?LOG_ERROR([{event, pool_failed_to_validate_proof_of_access}]),
+					#partial_solution_response{ status = <<"rejected_bad_poa">> };
+				false ->
+					#partial_solution_response{ status = <<"rejected_bad_poa">> };
+				{true, Chunk2ID} ->
+					PoA2Cache = {{BlockStart2, RecallByte2, TXRoot2, BlockSize2,
+							{spora_2_6, MiningAddress}}, Chunk2ID},
+					PoACache = {{BlockStart1, RecallByte1, TXRoot1, BlockSize1,
+							{spora_2_6, MiningAddress}}, ChunkID},
+					process_partial_solution_difficulty(Solution, Ref, PoACache, PoA2Cache)
+			end
+	end.
+
+process_partial_solution_difficulty(Solution, Ref, PoACache, PoA2Cache) ->
+	#mining_solution{ solution_hash = SolutionH } = Solution,
+	Diff = ar_node:get_current_diff(),
+	case binary:decode_unsigned(SolutionH, big) > Diff of
+		false ->
+			#partial_solution_response{ status = <<"accepted">> };
+		true ->
+			process_partial_solution_vdf(Solution, Ref, PoACache, PoA2Cache)
+	end.
+
+process_partial_solution_vdf(Solution, Ref, PoACache, PoA2Cache) ->
+	#mining_solution{
+		step_number = StepNumber,
+		next_seed = NextSeed,
+		start_interval_number = StartIntervalNumber,
+		next_vdf_difficulty = NextVDFDifficulty,
+		nonce_limiter_output = Output,
+		seed = Seed,
+		partition_upper_bound = PartitionUpperBound
+	} = Solution,
+	case ar_nonce_limiter:get_step_checkpoints_seed_upper_bound(StepNumber, NextSeed,
+			StartIntervalNumber, NextVDFDifficulty) of
+		not_found ->
+			#partial_solution_response{ status = <<"rejected_vdf_not_found">> };
+		{[Output | _] = LastStepCheckpoints, Seed, PartitionUpperBound} ->
+			Solution2 =
+				Solution#mining_solution{
+					last_step_checkpoints = LastStepCheckpoints,
+					%% ar_node_worker will fetch the required steps based on the prev block.
+					steps = not_found
+				},
+			ar_events:send(miner, {found_solution, {pool, Ref},
+					Solution2, PoACache, PoA2Cache}),
+			noreply;
+		_ ->
+			#partial_solution_response{ status = <<"rejected_bad_vdf">> }
```

Comment:
Ditto above: isn't this a valid partial solution?

Reply:
Added a clarifying comment